### PR TITLE
[Merged by Bors] - feat: the generators of a presented group generate the presented group

### PR DESCRIPTION
**Pull request by trivial1711**
- Add `FreeGroup.closure_range_of`: the subgroup closure of the range of `FreeGroup.of : α → FreeGroup α` is `⊤`. (That is, the generators of a free group generate the free group.)
- Add `PresentedGroup.closure_range_of`: the subgroup closure of the range of `PresentedGroup.of : α → PresentedGroup rels` is `⊤`. (That is, the generators of a presented group generate the presented group.)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/FreeGroup/Basic.lean [hunk 1]
```diff
@@ -777,6 +781,15 @@ theorem lift.range_eq_closure : (lift f).range = Subgroup.closure (Set.range f)
 #align free_group.lift.range_eq_closure FreeGroup.lift.range_eq_closure
 #align free_add_group.lift.range_eq_closure FreeAddGroup.lift.range_eq_closure
 
+/-- The generators of `FreeGroup α` generate `FreeGroup α`. That is, the subgroup closure of the
+set of generators equals `⊤`. -/
+@[to_additive]
```
Comment (ocfnash) [inline Mathlib/GroupTheory/FreeGroup/Basic.lean:786]:
```suggestion
@[to_additive (attr := simp)]
```

File: Mathlib/GroupTheory/PresentedGroup.lean [hunk 2]
```diff
@@ -47,6 +47,15 @@ def of {rels : Set (FreeGroup α)} (x : α) : PresentedGroup rels :=
   QuotientGroup.mk (FreeGroup.of x)
 #align presented_group.of PresentedGroup.of
 
+/-- The generators of a presented group generate the presented group. That is, the subgroup closure
+of the set of generators equals `⊤`. -/
+theorem closure_range_of (rels : Set (FreeGroup α)) :
```
Comment (ocfnash) [inline Mathlib/GroupTheory/PresentedGroup.lean:52]:
```suggestion
@[simp]
theorem closure_range_of (rels : Set (FreeGroup α)) :
```

File: Mathlib/GroupTheory/FreeGroup/Basic.lean [hunk 3]
```diff
@@ -777,6 +781,15 @@ theorem lift.range_eq_closure : (lift f).range = Subgroup.closure (Set.range f)
...
+theorem closure_range_of (α) :
+    Subgroup.closure (Set.range (FreeGroup.of : α → FreeGroup α)) = ⊤ := by
+  rw [← lift.range_eq_closure, lift_of_eq_id]
+  apply MonoidHom.range_top_of_surjective
+  exact fun x ↦ ⟨x, rfl⟩
```
Comment (ocfnash) [inline Mathlib/GroupTheory/FreeGroup/Basic.lean:791]:
```suggestion
  exact MonoidHom.range_top_of_surjective _ Function.surjective_id
```

**Review by ocfnash — APPROVED** [review]
Thanks!

bors d+

**Comment by trivial1711** [comment]
Thank you so much!

**Pull request closed without merging**